nf_core/modules.py

```diff
@@ -231,3 +240,259 @@ def has_valid_pipeline(self):
+                        "[violet]Test tags[/] (comma separated)", default=",".join(tag_defaults)
+                    ).strip()
+                    ep_test["tags"] = [t.strip() for t in prompt_tags.split(",")]
+
+        ep_test["files"] = self.get_md5_sums(entry_point, ep_test["command"])
+
+        return ep_test
+
+    def _md5(self, fname):
+        """Generate md5 sum for file"""
+        hash_md5 = hashlib.md5()
+        with open(fname, "rb") as f:
+            for chunk in iter(lambda: f.read(4096), b""):
+                hash_md5.update(chunk)
+        md5sum = hash_md5.hexdigest()
+        return md5sum
+
+    def get_md5_sums(self, entry_point, command):
+        """
+        Recursively go through directories and subdirectories
+        and generate tuples of (<file_path>, <md5sum>)
+        returns: list of tuples
+        """
+
+        results_dir = None
+        run_this_test = False
+        while results_dir is None:
+            if self.run_tests or run_this_test:
+                results_dir = self.run_tests_workflow(command)
+            else:
+                results_dir = rich.prompt.Prompt.ask(
+                    f"[violet]Test output folder with results[/] (leave blank to run test)"
+                )
+                if results_dir == "":
+                    results_dir = None
+                    run_this_test = True
+                elif not os.path.isdir(results_dir):
+                    log.error(f"Directory '{results_dir}' does not exist")
+                    results_dir = None
+
+        test_files = []
+        for root, dir, file in os.walk(results_dir):
+            for elem in file:
+                elem = os.path.join(root, elem)
+                elem_md5 = self._md5(elem)
+                # Switch out the results directory path with the expected 'output' directory
+                elem = elem.replace(results_dir, "output")
+                test_files.append({"path": elem, "md5sum": elem_md5})
+
+        if len(test_files) == 0:
+            raise UserWarning(f"Could not find any test result files in '{results_dir}'")
+
+        return test_files
+
+    def run_tests_workflow(self, command):
+        """ Given a test workflow and an entry point, run the test workflow """
+
+        # The config expects $PROFILE and Nextflow fails if it's not set
+        if os.environ.get("PROFILE") is None:
+            log.debug("Setting env var '$PROFILE' to an empty string as not set")
+            os.environ["PROFILE"] = ""
+
+        tmp_dir = tempfile.mkdtemp()
+        command += f" --outdir {tmp_dir}"
+
+        log.info(f"Running '{self.module_name}' test with command:\n[violet]{command}")
+        try:
+            nfconfig_raw = subprocess.check_output(shlex.split(command))
+        except OSError as e:
+            if e.errno == errno.ENOENT:
+                raise AssertionError(
+                    "It looks like Nextflow is not installed. It is required for most nf-core functions."
+                )
```

Comment:
I got here because I accidentally typed in something for the command. Usually nobody will do that but maybe we make this message more generic to indicate that something is off with the command? Should still point to nextflow though

Reply:
Ah good point, I think it's probably better to just rework this to catch all exceptions. This type is for "file not found" and I stole it from the configs code where we are building the command so know it's calling Nextflow. But yeah can just make this more generic so that it catches everything that's not a failed nextflow run.

Reply:
Ok left this as it is, but added a check so that it only triggers if the command started with `nextflow `. Added a catch-all exception for everything else.